src/Symfony/Component/HttpKernel/UriSigner.php

```diff
@@ -106,4 +134,22 @@ private function buildUrl(array $url, array $params = []): string
 
         return $scheme.$user.$pass.$host.$port.$path.$query.$fragment;
     }
+
+    private function getExpirationTime(\DateTimeInterface|\DateInterval|int $expiration): string
```

Comment:
We should leverage `DatePoint` here

https://github.com/symfony/symfony/pull/51415

Reply:
You want me to just replace `\DateTimeInterface` by `DatePoint` ?

Reply:
Best way would be to add it to the union type, no?

Reply:
`DatePoint` extends `\DateTimeImmutable` so it is also a `\DateTimeInterface`. Should we really add it to the union type ?
IMO it is not necessary.